pkg/util/httpproxy/upgradeawareproxy.go

```diff
@@ -183,6 +200,8 @@ func (p *UpgradeAwareSingleHostReverseProxy) serveUpgrade(w http.ResponseWriter,
 	}()
 
 	go func() {
+		// TODO: Intercept a 401 response, log, and transform into a 500 internal server error
```

Comment:
Sorry to make you move this again, but it really belongs after the request is written to the backendConn but before the goroutines are started